theano/gof/opt.py

```diff
@@ -2053,12 +2014,39 @@ def print_profile(stream, prof, level=0):
         print(blanc, "  loop time", loop_t, file=stream)
         print(blanc, "  callback_time", callback_time, file=stream)
         if isinstance(lopt, LocalOptGroup):
-            lopt.print_profile(stream, (lopt.time_opts,
```

Comment:
I'm not able to see this comment as i think you had commented on the page
which was open in your browser and after I rebased, github doesn't show
this comment.
I currently don't return anything from that method. I think I have found
the error. I had forgotten to remove the time_nodes method and have
initialized the keys of the dict with opts and tried to access with the key
as node. That could be the problem. But because of the bad rebase i'm not
able to figure that out. I will do the rebase again and push that change.
That change has been done in
https://github.com/Sentient07/Theano/tree/fixed-cgt-opt.

ᐧ

On Thu, Sep 1, 2016 at 4:35 AM, Frédéric Bastien <notifications@github.com>
wrote:

> In theano/gof/opt.py
> <https://github.com/Theano/Theano/pull/4876#discussion_r77106359>:
>
> > @@ -2053,12 +2014,39 @@ def print_profile(stream, prof, level=0):
> >          print(blanc, "  loop time", loop_t, file=stream)
> >          print(blanc, "  callback_time", callback_time, file=stream)
> >          if isinstance(lopt, LocalOptGroup):
> > -            lopt.print_profile(stream, (lopt.time_opts,
>
> I don't think having the print_profile can be the problem. But maybe
> keeping the results of profiling. Can you modify the method the execute the
> opt to return None instead of the profiling information? This will remove
> the information saved. If this is the extra information that raise the
> memory usage, then we will see the cause.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Theano/Theano/pull/4876/files/884c4989e1c4304f66f0246fe5807e37da3539b7..4aef3e2289a9fa394e3a32df31a3f69cfb4ec4c4#r77106359>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AF6IcKUrnX2RPL5A3oGkvwt3iD13oxL9ks5qljnzgaJpZM4JndrW>
> .
>